src/js/components/Drop/DropContainer.js

```diff
@@ -166,32 +166,27 @@ const DropContainer = forwardRef(
           let bottom;
           let maxHeight = containerRect.height;
           if (align.top) {
-            if (align.top === 'top') {
-              ({ top } = targetRect);
-            } else {
-              top = targetRect.bottom;
-            }
-
-            // Calculate visible area underneath the control w.r.t window height
-            const percentVisibleAreaBelow =
-              100 - (targetRect.bottom / windowHeight) * 100;
+            const alignToTop = align.top === 'top';
+            const switchTarget = !alignToTop
+              ? targetRect.top
+              : targetRect.bottom;
+            top = alignToTop ? targetRect.top : targetRect.bottom;
 
-            // Check whether it is within 20% from bottom of the window or
-            // visible area to flip the control
-            // DropContainer doesn't fit well within visible area when
-            // percentVisibleAreaBelow value<=20%
-            // There is enough space from DropContainer to bottom of the window
-            // when percentVisibleAreaBelow>20%.
-
-            if (windowHeight === top || percentVisibleAreaBelow <= 20) {
+            /* If responsive is true and the Drop doesn't have enough room 
+            to be fully visible and there is more room in the other 
+            direction, change the Drop to display above/below. If there is 
+            less room in the other direction leave the Drop in its current 
+            position. */
+            if (
+              responsive &&
+              (windowHeight === top ||
+                (windowHeight < top + containerRect.height &&
+                  windowHeight - top < switchTarget))
+            ) {
               // We need more room than we have.
               // We put it below, but there's more room above, put it above
               top = '';
-              if (align.top === 'bottom') {
-                bottom = targetRect.top;
-              } else {
-                ({ bottom } = targetRect);
-              }
+              bottom = alignToTop ? targetRect.bottom : targetRect.top;
               maxHeight = bottom;
               container.style.maxHeight = `${maxHeight}px`;
```

Comment:
How come sometimes we set `container.style.maxHeight` without honoring `preserveHeight`? I was expecting we'd only set the style within `if (!preserveHeight) {` later down.